Add growth engineer mentions in product engineer posts (#5867)
PostHog/posthog.com: contents/blog/what-is-a-product-engineer.md

Product engineers ideate new ways to make the product better. They then iterate on these ideas for improving the product and do it fast. This means a lot of experiments and tests. Tools like A/B tests and feature flags help with this.

> > **Product engineer vs growth engineer:** Product engineers experiment to improve a specific product they own. Growth engineers experiment at a higher level, focusing on increasing growth metrics like signups, subscriptions, and revenue across products. [Learn more about growth engineers in our post about them](/blog/what-is-a-growth-engineer).

Reviewer: ```suggestion
> **Product engineers vs growth engineers:** Product engineers experiment to improve a specific product they own. Growth engineers experiment at a higher level, focusing on increasing growth metrics like signups, subscriptions, and revenue across products. Read our [guide to growth engineers](/blog/what-is-a-growth-engineer) and why they're awesome to learn more.
```
